driver/src/contracts/stablex_auction_element.rs

```diff
@@ -68,19 +68,33 @@ impl StableXAuctionElement {
 fn compute_buy_sell_amounts(
     numerator: u128,
     denominator: u128,
-    amount: u128,
+    remaining: u128,
     is_sell_order: bool,
 ) -> (u128, u128) {
-    if denominator > 0 {
-        let other = (numerator * amount) / denominator;
-        if is_sell_order {
-            (other, amount)
-        } else {
-            (amount, other)
+    // TODO - use u128_to_u256 for overflow handling (this is a conversion used in naive_solver)
+    //    assert!(remaining <= denominator);
+    let mut other = 0;
+    let (buy_amount, sell_amount) = if is_sell_order {
+        // 0 on sellAmount (remaining <= denominator) is nonsense, but solver can handle it.
+        // 0 on buyAmount (numerator) is a Market Sell Order.
+        if denominator > 0 {
+            let p = remaining * numerator;
+            let q = denominator;
+            // Recall that ceil(p / float(q)) == (p + q - 1) / q
+            other = (p + q - 1) / q;
         }
+        (other, remaining)
     } else {
+        error!("Unable to to handle buy orders, returning unfulfillable amounts");
         (0, 0)
-    }
+        //        if numerator > 0 {
```

Comment:
Nice, yes, let's do not yet implement it. Leaving as comments is the best way right now

Reply:
I feel leaving unused and unfinished code in the codebase is unclean and makes it harder to understand what's going on. Is it code that would be hard to rewrite? it's also in version control as part of this PR (a comment with a link to a commit that contains the code could suffice instead of leaving it here).

Reply:
> Is it code that would be hard to rewrite?

It should not be hard to write, but perhaps we shouldn't bother until we have buy Orders.

> it's also in version control as part of this PR (a comment with a link to a commit that contains the code could suffice instead of leaving it here).

Actually this is not true, since our merges are squashed and branches deleted.

In any case, I am fine to take it out since it is incorrect in its current state anyway.

Reply:
> Actually this is not true, since our merges are squashed and branches deleted.

Merges are squashed into one commit on master the original commits remain. Branches are just labels for commits, so it doesn't matter if they are deleted.

E.g. this link points at a shell script of my rustfmt PR: https://github.com/gnosis/dex-services/pull/246/commits/7b78c74be4426062520078bca2de3e9cc7fd2be8#diff-b81e3e3f4c6ef7d2b9b3c06990d4cc17

The code got removed, the commit squash-merged and the branch deleted. Yet the code is still there.

Reply:
Interesting, thanks for that!